wordpress-mobile/WordPress-iOS
# WordPress to Jetpack Migration Welcome Screen

Fixes #19510

## Description
Adds the ‘Welcome Screen’, which is the first screen of the Migration Flow.

| Light | Dark |
| ---- | ------ |
| ![image](https://github.com/wordpress-mobile/WordPress-iOS/assets/9609223/5b58622d-ec07-4915-8df8-9f1c684d47d5) | ![image](https://github.com/wordpress-mobile/WordPress-iOS/assets/9609223/98a15096-ac41-4cfb-8680-be0a94928057) |

## Changes
| Change | Notes | Files | Screenshot |
| ------- | ------- | ---- | ------------ |
| Actions Sheet  | Reusable view that appear at the bottom of the Migration screens. | [MigrationActionsView.swift](https://github.com/wordpress-mobile/WordPress-iOS/pull/19533#diff-4c24a5fffb9459c9b558093950fb5df343057c0b5bba3fba263376ec693506aa) | <img width="250" alt="actions-sheet" src="https://user-images.githubusercontent.com/9609223/199561559-cf45edaa-4839-4d46-accd-48b6832adbaf.png"> |
| Header | Reusable view that appear at the top of the Migration screens. | [MigrationHeaderView.swift](https://github.com/wordpress-mobile/WordPress-iOS/pull/19533#diff-5af2d6618c9c4ae0f74cd749485e4c5d947422b1b96f3cf7a8bc9f86c93717f7) | <img width="250" alt="header" src="https://user-images.githubusercontent.com/9609223/199562209-14f05883-bdd5-4a25-9066-2092eb4105a5.png"> |
| "Me" Button  | Modified the existing component code, and it should be tested for regression. See **"'Me' Gravatar Button Regression Tests"** section. | [UIBarButtonItem+MeBarButton.swift](https://github.com/wordpress-mobile/WordPress-iOS/pull/19533#diff-0adecdb95492cde4e691510d54b24aa1df7deca52b2193f84496be9b2929fd59) | <img width="250" alt="me-button" src="https://user-images.githubusercontent.com/9609223/199562001-977dac43-a010-4da9-aa1a-6d05a6a3dd70.png"> |
| Welcome Screen  | The first step in the Migration flow. **"Continue" and "Need help?" are expected to do nothing for now.** | [MigrationWelcomeViewController.swift](https://github.com/wordpress-mobile/WordPress-iOS/pull/19533#diff-abe4968c501073aad9d7ce4b9221a6c10e0198052b7e520beb1ced44f1b84fcd) [MigrationWelcomeViewModel.swift](https://github.com/wordpress-mobile/WordPress-iOS/pull/19533#diff-962f3eccefaa26a6bea961ec5a100ec8f32b27fb048dc35c5ff9a672d1164995) [MigrationWelcomeBlogTableViewCell.swift](https://github.com/wordpress-mobile/WordPress-iOS/pull/19533#diff-39beb1861d5e244a60c1e366ee15091ae570c080f085a2ef74d3430ab80ddd5e)  </br> [UITableView+Header.swift](https://github.com/wordpress-mobile/WordPress-iOS/pull/19533#diff-95c8bb193134522ef8a9c83b319c664d9c0f094de9c216ad8608f34ec2953c3f) | N/A |
| Migration Coordinator | Acts as the source of truth for the migration flow, and it decides which `UIViewController` to display for each migration step. | [MigrationFlowCoordinator.swift](https://github.com/wordpress-mobile/WordPress-iOS/pull/19533#diff-0f0b0a8d2773baf849ccc0dfec84b0576af224ce5eb6ec2ebaa0526c413f7bd9) [MigrationFlowNavigationController.swift](https://github.com/wordpress-mobile/WordPress-iOS/pull/19533#diff-8bc8a184c81b1fc6fd73225325992c6f06284245b3fd546b0a7ffbddc99aebed) </br> [MigrationStep.swift](https://github.com/wordpress-mobile/WordPress-iOS/pull/19533#diff-bdd90d4487e10de848c1c2b83d13958eb1846ac935d39b2d0316fd2ded687eed) </br> [MigrationStepViewModel.swift](https://github.com/wordpress-mobile/WordPress-iOS/pull/19533#diff-476abdb523080b5d3a3d3aed05c1c5b4df670be856528519d61c1fc83e4caa2e) | N/A |

## Test Instructions

<details><summary><b>Prerequisites:</b>  Follow these steps to see the Welcome Screen when the Jetpack app launches</summary>

1. Locate the `showUI(for:)` method in `JetpackWindowManager` class.
2. Change its implementation to this:
```swift
class JetpackWindowManager: WindowManager {
    override func showUI(for blog: Blog?) {
        // If the user is logged in and has blogs sync'd to their account
        if AccountHelper.isLoggedIn && AccountHelper.hasBlogs {
            let context = ContextManager.shared.mainContext
            let account = try! WPAccount.lookupDefaultWordPressComAccount(in: context)
            let coordinator = MigrationFlowCoordinator(account: account!)
            self.show(coordinator.navigationController, completion: nil)
//            showAppUI(for: blog)
            return
        }

        // Show the sign in UI if the user isn't logged in
        guard AccountHelper.isLoggedIn else {
            showSignInUI()
            return
        }

        // If the user doesn't have any blogs, but they're still logged in, log them out
        // the `logOutDefaultWordPressComAccount` method will trigger the `showSignInUI` automatically
        AccountHelper.logOutDefaultWordPressComAccount()
    }
}
```
This is temporary and we'll eventually add a feature flag to force display the Migration screen when enabled.
</details>

#### Welcome Screen Test Instructions

1. Build and install Jetpack app.
2. **If not logged in:** Login then close & re-open the app
3. **Expect** to see the Welcome Screen.
4. **Verify** the Welcome Screen matches the design.
5. Tapping "Continue" or "Need help?" is expected to do nothing, because this part is still WIP.

#### "Me" Gravatar Button Regression Tests

1. Revert `JetpackWindowManager` changes
2. Build and install Jetpack app.
3. Log into your account if it's not already the case.
4. On "My Site" tab, tap the "Me" avatar button at the top.
5. **Expect** to see the User Account screen

## Regression Notes

1. Potential unintended areas of impact
The code behind "Me" avatar button was modified. So the button should be tested to make sure its behavior wasn't impacted.

3. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested "Me" button behavior.

4. What automated tests I added (or what prevented me from doing so)
Will add Unit Tests for Welcome Screen in future PRs.

## PR submission checklist:
